include/core/CSmallVector.h

```diff
@@ -96,7 +97,8 @@ class CSmallVector : public boost::container::small_vector<T, N> {
     //@{
     CSmallVector() {}
     CSmallVector(const CSmallVector& other) : TBase(other) {}
-    CSmallVector(CSmallVector&& other) : TBase(std::move(other.baseRef())) {}
+    CSmallVector(CSmallVector&& other) noexcept(std::is_nothrow_move_assignable<TBase>::value)
```

Comment:
nit: `std::is_nothrow_move_constructible` for the constructor (not that it will ever return a different value to `std::is_nothrow_move_assignable` in all likelihood)

Reply:
See [1788a8b](https://github.com/elastic/ml-cpp/pull/1149/commits/1788a8b7ce801b3b06c7cef07e037f789d255e52).